app/src/main/java/id/rizmaulana/covid19/ui/adapter/ItemTypeFactory.kt

```diff
@@ -24,4 +25,35 @@ class ItemTypeFactoryImpl: ItemTypeFactory() {
             else -> onCreateViewHolder(containerView, viewType)
         }
     }
+
+    override fun type(item: BaseViewItem): Int = when(item) {
+        is DailyItem -> DailyItemViewHolder.LAYOUT
+        is OverviewItem -> OverviewItemViewHolder.LAYOUT
+        is TextItem -> TextItemViewHolder.LAYOUT
+        is PinnedItem -> PinnedItemViewHolder.LAYOUT
+        is LocationItem -> LocationItemViewHolder.LAYOUT
+        is LoadingStateItem -> LoadingStateItemViewHolder.LAYOUT
+        is ErrorStateItem -> ErrorStateItemViewHolder.LAYOUT
+        else -> throw ClassCastException()
+    }
+
+}
+
+class DailyFactoryImpl: ItemTypeFactory() {
+    override fun onCreateViewHolder(
+        containerView: View,
+        viewType: Int
+    ): BaseViewHolder<out BaseViewItem> {
+        return when(viewType) {
+            DailyCompactViewHolder.LAYOUT -> DailyCompactViewHolder(containerView)
+            else -> throw ClassCastException()
+        }
+    }
+
+    override fun type(item: BaseViewItem): Int {
+        return when(item) {
+            is DailyItem -> DailyCompactViewHolder.LAYOUT
+            else -> throw ClassCastException()
+        }
+    }
```

Comment:
I see so we create a factory with ViewHolders that we only need with ItemTypeFactory as base, with this the factory wont become bloated Nice @fajarnuha,